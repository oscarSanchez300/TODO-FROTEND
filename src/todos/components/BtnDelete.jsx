import PropTypes from 'prop-types';


export const BtnDelete = ({id, loading, handleRemoveTodo}) => {
  return (
    <button 
        className="btn btn-sm btn-outline-danger" 
        onClick={() => handleRemoveTodo(id)}
        disabled={loading}
        data-testid="test-delete"
        role="dialog"
        aria-label='btn-delete'
    >
        Delete
    </button>
  )
}

BtnDelete.propTypes = {
  id: PropTypes.number.isRequired,
  loading: PropTypes.bool.isRequired,
  handleRemoveTodo: PropTypes.func.isRequired,
}
