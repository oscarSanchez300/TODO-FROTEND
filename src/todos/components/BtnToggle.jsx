
import PropTypes from "prop-types";

export const BtnToggle = ({todo, loading, handleToggleTodo}) => {
  return (
    <button 
        className="btn btn-sm btn-outline-success me-2" 
        onClick={() => handleToggleTodo(todo.id)}
        disabled={loading}
    >
        {todo.completed ? 'Undo' : 'Complete'}
    </button>
  )
}


BtnToggle.propTypes = {
  todo: PropTypes.object.isRequired,
  loading: PropTypes.bool.isRequired,
  handleToggleTodo: PropTypes.func.isRequired,
}
