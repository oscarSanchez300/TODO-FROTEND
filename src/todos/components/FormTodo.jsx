import React from "react";

export const FormTodo = React.memo(({handleAddTodo, loading}) => {


  const handleNewTodo = (e) => {
    e.preventDefault(); 

    // Create a new todo object    
    const newTodo = {
      text: e.target.todo.value,
      completed: false
    } 

    handleAddTodo(newTodo);
    
    e.target.todo.value = '';   
  }

  return (
    <form onSubmit={handleNewTodo} className="mt-4">
      <div className="input-group">
          <input 
              type="text" 
              name="todo" 
              className="form-control" 
              placeholder="Add a new todo" 
              required
          />
      </div>
        <button 
            className="btn btn-outline-primary mt-4" 
            type="submit"
            disabled={loading}
        >
          Add Todo
        </button>
    </form>
  )
})
