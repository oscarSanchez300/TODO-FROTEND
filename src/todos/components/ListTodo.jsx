import PropTypes from 'prop-types'
import { BtnDelete } from "./BtnDelete"
import { BtnToggle } from "./BtnToggle"

export const ListTodo = ({todo, loading, handleRemoveTodo, handleToggleTodo}) => {
  return (
    <li className="list-group-item d-flex justify-content-between align-items-center">
        <span className={todo.completed ? 'text-decoration-line-through' : ''}>
            {todo.text}
        </span>
        <div>
            <BtnToggle todo={todo} loading={loading} handleToggleTodo={handleToggleTodo}/>
            <BtnDelete id={todo.id} loading={loading} handleRemoveTodo={handleRemoveTodo}/>
        </div>
    </li>
  )
}


ListTodo.propTypes = {
  todo: PropTypes.object.isRequired,
  loading: PropTypes.bool.isRequired,
  handleRemoveTodo: PropTypes.func.isRequired,
  handleToggleTodo: PropTypes.func.isRequired,
}