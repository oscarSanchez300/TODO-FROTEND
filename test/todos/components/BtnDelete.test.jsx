import { fireEvent, render, screen } from "@testing-library/react";
import { BtnDelete } from "../../../src/todos/components/BtnDelete";



describe('Test on <BtnDelete />', () => { 
    
    const id = 53;
    const handleRemoveTodo = () => {};
    const loading = true;
    
    test('should of match with snapshot', () => {
        
        const { container } = render( <BtnDelete  id={id} loading={loading} handleRemoveTodo={handleRemoveTodo} /> );
        expect( container ).toMatchSnapshot();

    });

    test('should find text delete', () => {

        const { container, getByText, getByTestId } = render( <BtnDelete  id={id} loading={loading} handleRemoveTodo={handleRemoveTodo} /> );
        expect( getByTestId('test-delete').innerHTML ).toBe('Delete');
       
        // const btn = container.querySelector('button');
        // expect( getByText(btn.innerHTML) ).toBeTruthy();
        // expect( btn.innerHTML ).toBe('Delete');


    });

    test('Should find text delete with screen', () => {

        render( <BtnDelete id={id} loading={loading} handleRemoveTodo={handleRemoveTodo} /> );
        expect( screen.getByText('Delete') ).toBeTruthy();
        // screen.debug();

    });


    test('Should find text delete on button', () => {

        render( <BtnDelete id={id} loading={loading} handleRemoveTodo={handleRemoveTodo} /> );
        const dialogContainer = screen.getByRole('dialog');
        // console.log(dialogContainer.innerHTML);
        expect( dialogContainer.innerHTML ).toBe('Delete');
        // screen.debug();

    });

    test('should simulate event click on button Delete', () => { 

        render( <BtnDelete id={id} loading={loading} handleRemoveTodo={handleRemoveTodo} /> );
        fireEvent.click( screen.getByText('Delete') );
        fireEvent.click(screen.getByRole('dialog', { name: 'btn-delete' }));


        
    });

})