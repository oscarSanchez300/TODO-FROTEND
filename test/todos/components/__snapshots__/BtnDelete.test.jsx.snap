// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Test on <BtnDelete /> should of match with snapshot 1`] = `
<div>
  <button
    aria-label="btn-delete"
    class="btn btn-sm btn-outline-danger"
    data-testid="test-delete"
    disabled=""
    role="dialog"
  >
    Delete
  </button>
</div>
`;
